arch/x86/kvm/vmx/pkvm_host.c

```diff
@@ -238,6 +238,52 @@ static void pkvm_vcpu_free(struct kvm_vcpu *vcpu)
 	pkvm_free_loaded_vmcs(vmx->loaded_vmcs);
 }
 
+static void pkvm_vcpu_reset(struct kvm_vcpu *vcpu, bool init_event)
+{
+	struct vcpu_vmx *vmx = to_vmx(vcpu);
+
+	if (!pkvm_is_protected_vcpu(vcpu) && init_event)
+		pkvm_hypercall(vcpu_reset);
+
+	/*
+	 * The host is responsible for injecting interrupts to the guest. The
+	 * pi_desc is the key structure for the host to inject interrupts via
+	 * the posted interrupt mechanism. Its physical address is used for the
+	 * POSTED_INTR_DESC_ADDR in the VMCS by the pKVM hypervisor. Initialize
+	 * the pi_desc when reset vcpu.
+	 */
+	vmx->vt.pi_desc.nv = POSTED_INTR_VECTOR;
+	__pi_set_sn(&vmx->vt.pi_desc);
+
+	/*
+	 * The CR0/CR4 are emulated by the pKVM hypervisor. When the host reads
```

Comment:
nit: maybe s/emulated/managed/ (they are not fully emulated, some bits are passed-through to the guest)?